Parse/Parse.m

```diff
@@ -110,10 +132,7 @@ + (NSString *)getClientKey {
 + (void)enableDataSharingWithApplicationGroupIdentifier:(NSString *)groupIdentifier {
     PFConsistencyAssert(!currentParseManager_,
                         @"'enableDataSharingWithApplicationGroupIdentifier:' must be called before 'setApplicationId:clientKey'");
-    PFParameterAssert([groupIdentifier length], @"'groupIdentifier' should not be nil.");
```

Comment:
This validation doesn't exist anymore at this point. Probably should be brought back.